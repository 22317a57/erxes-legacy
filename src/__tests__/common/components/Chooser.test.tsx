import { mount, shallow } from 'enzyme';
import * as React from 'react';

import * as renderer from 'react-test-renderer';
import Chooser from '../../../modules/common/components/Chooser';

describe('Chooser component', () => {
  const defaultProps = {
    data: 'today',
    onSelect: (datas: any[]) => null,
    search: (value: string, reload?: boolean) => null,
    datas: ['datas'],
    title: 'title',
    renderName: (data: any) => null,
    renderForm: (props: { closeModal: () => void }) => 'any',
    perPage: 0,
    clearState: () => null,
    closeModal: () => null
  };

  test('renders Chooser successfully', () => {
    shallow(<Chooser {...defaultProps} />);
  });

  test('snapshot matches', () => {
    const rendered = renderer.create(<Chooser {...defaultProps} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  test('renders successfully with default value', () => {
    const wrapper = mount(<Chooser {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });
});
