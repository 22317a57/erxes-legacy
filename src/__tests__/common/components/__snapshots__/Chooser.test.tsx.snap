// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chooser component snapshot matches 1`] = `
<div>
  <div
    className="sc-hORach umMsy"
  >
    <div
      className="sc-bMVAic hypBaD"
    >
      <input
        className="sc-kaNhvL bKDVoH"
        disabled={false}
        onChange={[Function]}
        placeholder="Type to search"
        required={false}
      />
      <ul>
        <li
          onClick={[Function]}
        >
          <i
            className="icon-add sc-bwzfXH hpxbGw"
            icon="add"
          />
        </li>
        <div
          className="sc-kPVwWT duHBLj"
        >
          <button
            className="sc-ifAKCX bndCnC"
            icon="checked-1"
            onClick={[Function]}
            type="button"
          >
            <i
              className="icon-checked-1 sc-bwzfXH hpxbGw"
              icon="checked-1"
            />
            <span>
              Load More
            </span>
          </button>
        </div>
      </ul>
    </div>
    <div
      className="sc-bMVAic hypBaD"
    >
      <h4
        className="sc-bAeIUo geHemc"
      >
        's 
        title
        <span>
          (
          0
          )
        </span>
      </h4>
      <div
        className="sc-cugefK hrlmsg"
      >
        <i
          className="icon-list-2 sc-bwzfXH hpxbGw"
          icon="list-2"
        />
        <span>
          No items added
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-hwwEjo bdhUHy"
  >
    <div
      className="sc-iujRgT guNvFF"
    >
      <p
        onClick={[Function]}
      >
        Don't see the result you're looking for? 
        <a>
          Create a new title
        </a>
      </p>
      <div>
        <button
          className="sc-ifAKCX iSmYrG"
          icon="cancel-1"
          onClick={[Function]}
          type="button"
        >
          <i
            className="icon-cancel-1 sc-bwzfXH hpxbGw"
            icon="cancel-1"
          />
          <span>
            Cancel
          </span>
        </button>
        <button
          className="sc-ifAKCX cQdFDO"
          icon="checked-1"
          onClick={[Function]}
          type="button"
        >
          <i
            className="icon-checked-1 sc-bwzfXH hpxbGw"
            icon="checked-1"
          />
          <span>
            Select
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
